### ISSUE TITLE
QAOA Runtime method yielding inconsistent results

### ISSUE THREAD

**Issue by amitracal** [Qiskit/qiskit-optimization#281]
### Environment

- **Qiskit Optimization version**:
- **Python version**: 
- **Operating system**: 

![Version 11_21_21](https://user-images.githubusercontent.com/69091655/142808496-4f2abd49-4454-4b25-9433-0d3117a845fd.PNG)

### What is happening?

Results are inconsistent when using QAOAProgram method as shown in the attached notebook

### How can we reproduce the issue?

Run the attached notebook
[QAOA Runtime.zip](https://github.com/Qiskit/qiskit-optimization/files/7578567/QAOA.Runtime.zip)

### What should happen?

result should match with Numpy Minimum Eigen solver - global minima at 11100 optimized value 179

**Comment by t-imamichi** [comment]
Since QAOA is a heuristic algorithm, results of QAOA do not always match those of Numpy minimum eigen solver (exact algorithm).

**Comment by amitracal** [comment]
True but this is consistently inconsistent, which is not the case when I create a custom shell program for runtime on QAOA method instead of QAOAProgram method.

**Comment by t-imamichi** [comment]
@Cryoris Do you have any thought?

**Comment by Cryoris** [comment]
@amitracal The zip file you included doesn't work for me, could you check that it's the right file? Without more description it's a bit difficult to say more.

So far, I don't think there are known bugs with the VQE/QAOA programs. Are your reference results also from real devices or from simulations? How large are the circuits you're running?

**Comment by amitracal** [comment]
@Cryoris any updates ? I think you confirmed that you got the zip file.

**Comment by amitracal** [comment]
@Cryoris We have a quantum accelerator delivery for which this notebook is waiting for the defect https://github.com/Qiskit/qiskit-optimization/issues/281, do we have any ETA on this please ?

**Comment by Cryoris** [comment]
I've looked at your notebook @amitracal and I think there might be two reasons that you've seen different results locally and on the runtime:
* Locally you were using COBYLA with a fixed initial point and on the runtime QN-SPSA. These optimizers work very differently and they might not find the same minimum. Especially QN-SPSA can require some fine tuning. 
* Since you posted this issue in November we've release the QAOA runtime program and which might have fixed some issues we were unaware of.

I updated the notebook a bit to run COBYLA on the runtime too, with the same settings as locally, and I'm seeing the same results for both. Actually, also QN-SPSA does work locally with the settings I used (note: it doesn't work on the runtime right now, that's a bug with the new QAOA runtime program but shouldn't have been an issue with earlier versions!). So I think that everything works as expected (aside that the new QAOA program doesn't support QN-SPSA).

I've attached the notebook with my changes.
[qaoa_local_vs_runtime.zip](https://github.com/Qiskit/qiskit-optimization/files/7978417/qaoa_local_vs_runtime.zip)
